staging/src/k8s.io/client-go/transport/round_trippers.go

```diff
@@ -261,6 +268,54 @@ func (rt *impersonatingRoundTripper) CancelRequest(req *http.Request) {
 
 func (rt *impersonatingRoundTripper) WrappedRoundTripper() http.RoundTripper { return rt.delegate }
 
+// NewDebugConcurrentRoundTripper is meant to give track debug information about all currently
+// open transports and provide that information if something goes wrong.
+
+var debugConcurrentMutex sync.Mutex
+var debugConcurrentMap map[string]debugConcurrentRecord = make(map[string]debugConcurrentRecord)
```

Comment:
I would prefer to wrap these in an object. Even if there's a singleton instance of the object, at least that way you could declare a second one if you wanted to (for testing, say).